### PR TITLE
add HelloJsni sample to demonstrate JSNI support

### PR DESCRIPTION
**Pull request by maffoo**

### CODE REVIEW

**Comment by anovstrup** [comment]
The showcase is using Java in several places where jsni is needed. Can we replace these with Scala?

**Comment by gkossakowski** [comment]
@anovstrup: Good idea! Are you feeling like trying @maffoo's [patch](https://github.com/maffoo/scalagwt-gwt) and working on this idea? We are getting really close to 100% Scala samples :-)

**Comment by anovstrup** [comment]
sure, I'll take a stab at it tonight, and verify the patch.

On Wed, Apr 11, 2012 at 11:18 AM, Grzegorz Kossakowski
reply@reply.github.com
wrote:

**Comment by gkossakowski** [comment]
Cool. One caveat: I'd leave showcase as a last step. There's rewrite in the process https://github.com/gkossakowski/scalagwt-sample/commit/bb5a94f754ad0e9cc7ce4f0cd78ab19c47b0cc3a

**Comment by anovstrup** [comment]
Ok, in that case I'll wait on showcase until I get the green light
from you. I have more free time later in the week anyway.

On Wed, Apr 11, 2012 at 11:27 AM, Grzegorz Kossakowski
reply@reply.github.com
wrote:

File: src/com/google/gwt/sample/jsni/HelloJsni.gwt.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module rename-to='hellojsni'>
+  <inherits name='com.google.gwt.user.User'/>
+  <inherits name='ScalaLibrary'/>
+  <entry-point class='com.google.gwt.sample.jsni.client.HelloJsni'/>
+  <source path='client'/>
+  <set-property name='user.agent' value='safari'/>
```
Comment (anovstrup) [inline src/com/google/gwt/sample/jsni/HelloJsni.gwt.xml:7]: need to remove this line

File: src/com/google/gwt/sample/jsni/client/HelloJsni.scala [hunk 2]
```diff
@@ -0,0 +1,94 @@
+package com.google.gwt.sample.jsni.client
+
+//import scala.util.nativeCode
```
Comment (anovstrup) [inline src/com/google/gwt/sample/jsni/client/HelloJsni.scala:3]: let's remove this line if it's unneeded

**Comment by anovstrup** [comment]
Overall comments:
1. The clean build target needs to be updated.
2. The HelloJsni sample threw a runtime exception in production mode ("uncaught exception: java.lang.RuntimeException: nativeCode: method body not provided
   file:///home/aaron/scala-gwt/sample/war/hellojsni/DBC7A869382417F1641D7B1AC4F9A0B1.cache.html
   Line 678").  I compiled with the scala/gwt from the corresponding pull requests to scalagwt-scala and scalagwt-gwt.
3. I made a few minor code comments, but otherwise it looks fine.

**Pull request closed without merging**